src/mlpack/core/data/dictionary_encoding_impl.hpp

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file dictionary_encoding_impl.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of dictionary encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_DICT_ENCODING_IMPL_HPP
+#define MLPACK_CORE_DATA_DICT_ENCODING_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "dictionary_encoding.hpp"
+
+namespace mlpack {
+namespace data {
+
+void DicitonaryEncoding::Reset()
+{
+  mappings.clear();
+  stringq.clear();
+}
+
+DicitonaryEncoding::DicitonaryEncoding(const DicitonaryEncoding &old_obj)
+{
+  this->stringq = old_obj.stringq;
+  std::deque<std::string>::iterator jt = stringq.begin();
+  for (auto it = old_obj.mappings.begin(); it != old_obj.mappings.end(); it++)
+  {
+    this->mappings[*jt] = it->second;
+    jt++;
+  }
+}
+
+void DicitonaryEncoding::operator= (const
+    DicitonaryEncoding &old_obj)
+{
+  if (this != &old_obj)
+  {
+    this->mappings.clear();
+    this->stringq.clear();
+    this->stringq = old_obj.stringq;
+    std::deque<std::string>::iterator jt = stringq.begin();
+    for (auto it = old_obj.mappings.begin(); it != old_obj.mappings.end(); it++)
+    {
+      this->mappings[*jt] = it->second;
```

Comment:
Again, the ordering might be different. I think it's a good idea to write a test case for the constructor and for the copy-assignment operator.

Reply:
Since both of them follow almost same definition, I have wrote for one, and by generalization other will follow.